tests/conftest.py

```diff
@@ -62,6 +73,184 @@ def pytest_configure(config):
+            data = json.loads(fn.read_text())
+            data["finished"] += 1
+            fn.write_text(json.dumps(data))
+    else:
+        # TODO(yee): improve this option format, only specify it by `--stop_nebula`
+        stop_nebula = pytestconfig.getoption("stop_nebula")
+        while stop_nebula.lower() == "true":
+            data = json.loads(fn.read_text())
+            if data["finished"] + 1 == data["num_workers"]:
+                nb.stop()
+                break
+            time.sleep(1)
+        os.remove(str(fn))
+
+
+@pytest.fixture(scope="session")
+def session(conn_pool, pytestconfig):
+    user = pytestconfig.getoption("user")
+    password = pytestconfig.getoption("password")
+    sess = conn_pool.get_session(user, password)
+    yield sess
+    sess.release()
+
+
+def load_csv_data(pytestconfig, conn_pool, folder: str):
+    data_dir = os.path.join(CURR_PATH, 'data', folder)
+    schema_path = os.path.join(data_dir, 'schema.ngql')
+
+    user = pytestconfig.getoption("user")
+    password = pytestconfig.getoption("password")
+    sess = conn_pool.get_session(user, password)
+
+    with open(schema_path, 'r') as f:
+        stmts = []
+        for line in f.readlines():
+            ln = line.strip()
+            if ln.startswith('--'):
+                continue
+            stmts.append(ln)
+        rs = sess.execute(' '.join(stmts))
+        assert rs.is_succeeded()
+
+    time.sleep(3)
+
+    for path in Path(data_dir).rglob('*.csv'):
+        for stmt in CSVImporter(path):
+            rs = sess.execute(stmt)
+            assert rs.is_succeeded()
+
+    sess.release()
+
+
+# TODO(yee): optimize data load fixtures
+@pytest.fixture(scope="session")
+def load_nba_data(conn_pool, pytestconfig, tmp_path_factory, worker_id):
+    root_tmp_dir = tmp_path_factory.getbasetemp().parent
+    fn = root_tmp_dir / "csv-data-nba"
+    load = False
+    with FileLock(str(fn) + ".lock"):
+        if not fn.is_file():
+            load_csv_data(pytestconfig, conn_pool, "nba")
+            fn.write_text("nba")
+            logging.info(f"session-{worker_id} load nba csv data")
+            load = True
+        else:
+            logging.info(f"session-{worker_id} need not to load nba csv data")
+    yield
+    if load:
+        os.remove(str(fn))
+
+
+@pytest.fixture(scope="session")
+def load_student_data(conn_pool, pytestconfig, tmp_path_factory, worker_id):
+    root_tmp_dir = tmp_path_factory.getbasetemp().parent
+    fn = root_tmp_dir / "csv-data-student"
+    load = False
+    with FileLock(str(fn) + ".lock"):
+        if not fn.is_file():
+            load_csv_data(pytestconfig, conn_pool, "student")
+            fn.write_text("student")
+            logging.info(f"session-{worker_id} load student csv data")
+            load = True
+        else:
+            logging.info(
+                f"session-{worker_id} need not to load student csv data")
+    yield
+    if load:
+        os.remove(str(fn))
```

Comment:
you can define one function, and make the space_name as input args

Reply:
make sense, I am thinking how to optimize these fixtures if the `yield`sentence in different functions.